client/src/sensemakerStore.ts

```diff
@@ -410,6 +410,43 @@ export class SensemakerStore {
     return appletConfig;
   }
   
+  async registerWidget(widgetRegistration: AssessmentWidgetRegistrationInput): Promise<AssessmentWidgetRegistration> {
+    const result = await this.service.registerWidget(widgetRegistration);
+    return new EntryRecord<AssessmentWidgetRegistration>(result).entry
+  }
+
+  async getRegisteredWidgets(): Promise<Record<EntryHashB64, AssessmentWidgetRegistration>> {
+    const result : HolochainRecord[] = await this.service.getRegisteredWidgets();
+    return result.reduce((record, value) => {
+      const entryRecord = new EntryRecord<AssessmentWidgetRegistration>(value);
+      const key: EntryHashB64 = encodeHashToBase64(entryRecord.entryHash);
+      const registration: AssessmentWidgetRegistration = entryRecord.entry;
+      
+      record[key] = registration;
+      return record
+    }, {} as Record<EntryHashB64, AssessmentWidgetRegistration>)
+  }
+
+  // async registerAppletConfigWidgets(appletEh: EntryHash, assessmentWidgets: AssessmentWidgetConfigDict) {
```

Comment:
I'm going to merge this, but please never leave commented code in here like this. Also, Promise.all will not work in this situation as creating the entry changes the head of the source chain and multiple parallel writes will throw an error unless they changed the behavior in Holochain 0.2.x or 0.3.x. This is why I made the fiber object in the REA playspace to handle things serially, but I still need to change it to handle returning values and add better error handling: https://github.com/lightningrodlabs/rea-playspace/blob/main/modules/data-providers/src/lib/fiber.ts#L53

Additionally, this should probably be a zome function that accepts a BTreeMap so that way we can define better error handling.